front/src/modules/simulationResult/components/SpeedSpaceChart/useSpeedSpaceChart.ts

```diff
@@ -0,0 +1,216 @@
+
+  const [formattedPathProperties, setFormattedPathProperties] = useState<PathPropertiesFormatted>();
+  const [formattedPowerRestrictions, setFormattedPowerRestrictions] = useState<
+    SimulationPowerRestrictionRange[]
+  >([]);
+
+  const { data: rollingStock } =
+    osrdEditoastApi.endpoints.getRollingStockNameByRollingStockName.useQuery(
+      {
+        rollingStockName: trainScheduleResult?.rolling_stock_name as string,
+      },
+      { skip: !trainScheduleResult }
+    );
+
+  const { data: path } = osrdEditoastApi.endpoints.getV2TrainScheduleByIdPath.useQuery(
+    {
+      id: selectedTrainId as number,
+      infraId: infraId as number,
+    },
+    { skip: !selectedTrainId || !infraId }
+  );
+
+  const { data: trainSimulation } =
+    osrdEditoastApi.endpoints.getV2TrainScheduleByIdSimulation.useQuery(
+      { id: selectedTrainId as number, infraId: infraId as number },
+      { skip: !selectedTrainId || !infraId }
+    );
+
+  const [postPathProperties] =
+    osrdEditoastApi.endpoints.postV2InfraByInfraIdPathProperties.useMutation();
+
+  const simulation = trainSimulation || stdcmResponse?.simulation;
+  const pathfindingResult = path?.status === 'success' ? path : stdcmResponse?.path;
+  const departureTime =
+    trainScheduleResult?.id === STDCM_TRAIN_ID
+      ? stdcmResponse?.departure_time
+      : trainScheduleResult?.start_time;
+
+  // retrieve and format pathfinding properties
+  useEffect(() => {
+    const getPathProperties = async () => {
+      if (
+        infraId &&
+        trainScheduleResult &&
+        rollingStock &&
+        pathfindingResult &&
+        simulation?.status === 'success'
+      ) {
+        const pathPropertiesParams: PostV2InfraByInfraIdPathPropertiesApiArg = {
+          infraId,
+          props: ['electrifications', 'geometry', 'operational_points', 'curves', 'slopes'],
+          pathPropertiesInput: {
+            track_section_ranges: pathfindingResult.track_section_ranges,
+          },
+        };
+        const pathPropertiesResult = await postPathProperties(pathPropertiesParams).unwrap();
+
+        const formattedPathProps = preparePathPropertiesData(
+          simulation.electrical_profiles,
+          pathPropertiesResult,
+          pathfindingResult.length
+        );
+        setFormattedPathProperties(formattedPathProps);
+
+        // Format power restrictions
+        if (trainScheduleResult && trainScheduleResult.power_restrictions && rollingStock) {
+          const powerRestrictionsRanges = formatPowerRestrictionRanges(
+            trainScheduleResult.power_restrictions,
+            trainScheduleResult.path,
+            pathfindingResult.path_items_positions
+          );
+          const powerRestrictionsWithHandled = formatPowerRestrictionRangesWithHandled(
+            powerRestrictionsRanges,
+            formattedPathProps.electrifications,
+            rollingStock.effort_curves.modes
+          );
```

Comment:
And here in a separate handlePowerRestrictions function

Reply:
Done !